src/Formats/MySQLBlockInputStream.cpp

```diff
@@ -132,27 +135,69 @@ Block MySQLBlockInputStream::readImpl()
     for (const auto i : ext::range(0, columns.size()))
         columns[i] = description.sample_block.getByPosition(i).column->cloneEmpty();
 
+    if (unlikely(position_mapping.size() != description.sample_block.columns()))
```

Comment:
```suggestion
    if (unlikely(position_mapping.empty()))
```

Reply:
Btw, can we initialize `position_mapping` in constructor? And what will happen if query has returned 0 rows? Seems like we will not check block structure in this case.

Reply:
> Btw, can we initialize position_mapping in constructor? 

Done.

> And what will happen if query has returned 0 rows? Seems like we will not check block structure in this case.

return Block{} if query result structure is matched. Otherwise,  throw Exception.